File: testkit/src/main/resources/scenarios/java/OrderDate.java
package scenarios.java;

import com.qualityminds.lazyval.LazyValue;
import java.time.LocalDate;

@LazyValue
public record OrderDate(LocalDate value) {
    public OrderDate {
        if (value == null) {
            throw new IllegalArgumentException("OrderDate must not be null");
        }
        if(value.isAfter(LocalDate.now())){
            throw new IllegalArgumentException("OrderDate must not be in the future");
        }
    }
}

File: testkit/src/main/resources/scenarios/kotlin/OrderDate.kt
package scenarios.kotlin

import com.qualityminds.lazyval.LazyValue;
import java.time.LocalDate;

@LazyValue
data class OrderDate(val date: LocalDate){

    init {
        require(!date.isAfter(LocalDate.now())) { "OrderDate date cannot be in the future" }
    }
}